language/stdlib/modules/LibraVMConfig.move

```diff
@@ -3,15 +3,30 @@ address 0x1 {
 module LibraVMConfig {
     use 0x1::LibraConfig;
     use 0x1::Roles;
+    use 0x1::Vector;
+
+    const SCRIPT_HASH_LENGTH: u64 = 32;
+    const ESCRIPT_HASH_LENGTH_NOT_MATCH: u64 = 0;
 
     // The struct to hold all config data needed to operate the LibraVM.
     // * publishing_option: Defines Scripts/Modules that are allowed to execute in the current configruation.
     // * gas_schedule: Cost of running the VM.
     struct LibraVMConfig {
-        publishing_option: vector<u8>,
+        publishing_option: PublishingOption,
         gas_schedule: GasSchedule,
     }
 
+    struct PublishingOption {
+        // Only script hashes in the following list can be executed by the network.
+        script_allow_list: vector<vector<u8>>,
+        // VM can execute arbitrary script provided by the user if this flag is set to true.
+        is_open_script: bool,
```

Comment:
What did you think of @tzakian's suggestion to represent this with an empty `script_allow_list`?

Reply:
That would actually simplify the settings IMHO as long as this seems fine to you. I'm a bit hesitated just I view it as a wrapper for Option<Vec<HashValue>>, so I'm more inclined to use a boolean field. On the other hand I'm ok with the suggestion.

Reply:
Looking at the code it might be a bit trickier to change. The reason is that we have a smoke test suite to test the behavior from changing the network from arbitrary script execution to a limited mode. I think the bigger issue here is to come up with the state machine between aribitrary script execution and limited execution mode. Thought? @tzakian

Reply:
Wouldn't changing from arbitrary to limited mode just be adding a number of transactions to the allowlist? 

The way I view the state diagram is something like:
* STATE 1: open scripts
* STATE 2: only allowed scripts

We would then have the following transitions.

* In STATE 1:
   - Add script to allowlist ~> TRANSITION(STATE 2)
* in STATE 2:
   - Add script to allowlist ~> TRANSITION(STATE 2)
   - Remove script to allowlist ~> if EMPTY(allowlist) TRANSITION(STATE 1) else TRANSITION(STATE 2)